src/Compilers/CSharp/Desktop/CommandLine/CommandLineParser.cs

```diff
@@ -554,8 +554,16 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string bas
                                 {
                                     generalDiagnosticOption = ReportDiagnostic.Error;
 
-                                    // Clear specific warnaserror options (since last /warnaserror flag on the command line always wins).
+                                    // Reset specific warnaserror options (since last /warnaserror flag on the command line always wins),
+                                    // and bump warnings to errors.
                                     warnAsErrors.Clear();
+                                    foreach (var key in diagnosticOptions.Keys)
+                                    {
+                                        if (diagnosticOptions[key] == ReportDiagnostic.Warn)
```

Comment:
Didn't we say that in such cases, we also need to copy over all other values from the ruleset file?